src/test/tor/minimal/tor-minimal.yaml

```diff
@@ -30,114 +28,112 @@ topology:
-      -f ../../../conf/tor.exit.torrc --BandwidthRate 1024000 --BandwidthBurst 1024000
-- id: exit2
-  quantity: 1
-  bandwidthdown: 10240
-  bandwidthup: 10240
-  processes:
-  - plugin: tor
-    starttime: 60
-    arguments: --Address exit2 --Nickname exit2 --DataDirectory .
-      --GeoIPFile ~/.shadow/share/geoip --defaults-torrc ../../../conf/tor.common.torrc
-      -f ../../../conf/tor.exit.torrc --BandwidthRate 1024000 --BandwidthBurst 1024000
-- id: relay1
-  quantity: 1
-  bandwidthdown: 10240
-  bandwidthup: 10240
-  processes:
-  - plugin: tor
-    starttime: 60
-    arguments: --Address relay1 --Nickname relay1 --DataDirectory .
-      --GeoIPFile ~/.shadow/share/geoip --defaults-torrc ../../../conf/tor.common.torrc
-      -f ../../../conf/tor.relay.torrc --BandwidthRate 1024000 --BandwidthBurst 1024000
-- id: relay2
-  quantity: 1
-  bandwidthdown: 10240
-  bandwidthup: 10240
-  processes:
-  - plugin: tor
-    starttime: 60
-    arguments: --Address relay2 --Nickname relay2 --DataDirectory .
-      --GeoIPFile ~/.shadow/share/geoip --defaults-torrc ../../../conf/tor.common.torrc
-      -f ../../../conf/tor.relay.torrc --BandwidthRate 1024000 --BandwidthBurst 1024000
-- id: client
-  processes:
-  - plugin: tgen
-    starttime: 600
-    arguments: ../../../conf/tgen.client.graphml.xml
-- id: torclient
-  processes:
-  - plugin: tor
-    starttime: 900
-    arguments: --Address torclient --Nickname torclient --DataDirectory .
-      --GeoIPFile ~/.shadow/share/geoip --defaults-torrc ../../../conf/tor.common.torrc
-      -f ../../../conf/tor.client.torrc --BandwidthRate 1024000 --BandwidthBurst 1024000
-  - plugin: tgen
-    starttime: 1500
-    arguments: ../../../conf/tgen.torclient.graphml.xml
-- id: torbridgeclient
-  processes:
-  - plugin: tor
-    starttime: 900
-    arguments: --Address torbridgeclient --Nickname torbridgeclient --DataDirectory .
-      --GeoIPFile ~/.shadow/share/geoip --defaults-torrc ../../../conf/tor.common.torrc
-      -f ../../../conf/tor.client.torrc --BandwidthRate 1024000 --BandwidthBurst 1024000
-      --UseBridges 1 --Bridge 100.0.0.1:9111
-  - plugin: tgen
-    starttime: 1500
-    arguments: ../../../conf/tgen.torclient.graphml.xml
-- id: torhiddenclient
-  processes:
-  - plugin: tor
-    starttime: 900
-    arguments: --Address torhiddenclient --Nickname torhiddenclient --DataDirectory .
-      --GeoIPFile ~/.shadow/share/geoip --defaults-torrc ../../../conf/tor.common.torrc
-      -f ../../../conf/tor.client.torrc --BandwidthRate 1024000 --BandwidthBurst 1024000
-  - plugin: tgen
-    starttime: 1500
-    arguments: ../../../conf/tgen.torhiddenclient.graphml.xml
+  fileserver:
+    options:
+      bandwidth_down: 819200 Kibit
+      bandwidth_up: 819200 Kibit
+    processes:
+    - path: ~/.shadow/bin/tgen
+      args: ../../../conf/tgen.server.graphml.xml
+      start_time: 1
+  hiddenserver:
+    options:
+      bandwidth_down: 819200 Kibit
+      bandwidth_up: 819200 Kibit
```

Comment:
Should we just set this bandwidth as the default for everyone in the `host_defaults` part of the config? Then we wouldn't need to duplicate it for each host.

Or even better, the topology only contains a single node, and that node already has the correct bandwidth, so I think all the hosts will already get assigned that bandwidth even if we don't specify it as a host option or host_default option.

Reply:
Some hosts have bandwidths of 819200 kibibit (102400 KiB/s), some have 81920 kibibit (10240 KiB/s), clients have no defined bandwidths, and the topology node has a bandwidth of 81920 kibibit. Is there a reason why the original configuration had these hosts with different bandwidths? And why some hosts had bandwiths 10x greater than the topology node? Since this is just a test, it should be fine giving them all the same bandwidth, right?

Reply:
Right, I don't think the bandwidths matter much in this small test. I think they can all be the same. Not sure why they were different before - probably because I decided that many many years ago and never changed it.

Reply:
Changed the topology bandwidth to `1 Gbit` (rounded up from the ~800 Mbit that most hosts were using) and unset all of the other bandwidths.